ai_dungeon_cli/__init__.py

```diff
@@ -81,27 +53,94 @@ def display_splash():
 
 
 class AiDungeon:
-    def __init__(self, auth_token, prompt):
+    def __init__(self):
         self.terminal_size = shutil.get_terminal_size((80, 20))
         self.terminal_width = self.terminal_size.columns
-        self.prompt = prompt
-        self.prompt_iteration = 0
+
+        # Variables initialization
+        self.prompt = "> "
+        self.auth_token = None
+        self.email = None
+        self.password = None
+        self.prompt_iteration = 2
         self.stop_session = False
         self.user_id = None
         self.session_id = None
         self.public_id = None
         self.story_configuration = {}
         self.session = requests.Session()
+
+        # Start class configuration
+        self.init_configuration_file()
+
+    def update_session_auth(self):
         self.session.headers.update(
             {
                 # 'cookie': cookie,
-                "X-Access-Token": auth_token,
+                "X-Access-Token": self.auth_token,
             }
         )
 
+    def init_configuration_file(self):
+        cfg_file = "/config.yml"
+        cfg_file_paths = [
+            os.path.dirname(os.path.realpath(__file__)) + cfg_file,
+            os.getenv("HOME") + "/.config/ai-dungeon-cli" + cfg_file,
+        ]
+
+        did_read_cfg_file = False
+
+        for file in cfg_file_paths:
+            try:
+                with open(file, "r") as cfg_raw:
+                    cfg = yaml.load(cfg_raw, Loader=yaml.FullLoader)
+                    did_read_cfg_file = True
+            except IOError:
+                pass
+
+        if not did_read_cfg_file:
+            print("Missing config file at ", end="")
+            print(*cfg_file_paths, sep=", ")
+            raise FailedConfiguration
+
+        if (not exists(cfg, "auth_token")) and (
+            not (exists(cfg, "email")) and not (exists(cfg, "password"))
+        ):
+            self.print_sentences(
+                "Missing or empty authentication configuration. "
+                "Please register a token ('auth_token' key) or credentials ('email' / 'password')"
+            )
+            raise FailedConfiguration
+
+        if exists(cfg, "prompt"):
+            self.prompt = cfg["prompt"]
+
+        self.auth_token = cfg["auth_token"]
+        self.email = cfg["email"]
+        self.password = cfg["password"]
```

Comment:
Won't these risk to cause a `KeyError`?

Reply:
well if they wouldn't exist, they would be dropped at the earlier `if` section where it checks if they don't exist in the configuration file, it can only happen if only part of the data exists (e.g. if there is only `email `and `password `but the `auth_token `field is empty)

Reply:
if you want, you can just add a couple more if's, but i think that it shouldn't be necessary as the configuration file should have them all, even if they are empty.

Reply:
> as the configuration file should have them all, even if they are empty

I don't agree. That's not how optional config items should work.